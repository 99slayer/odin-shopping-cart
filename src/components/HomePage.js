import React, { useState, useEffect } from "react";
import "../styles/HomePage.css";

export const HomePage = () => {
  return (
    <div id="home-page">
      <ProductDisplay />
      <Reviews />
      <About />
    </div>
  );
};

const ProductDisplay = () => {
  return (
    <div id="product-display">
      <img alt="" src="toaster-1.jpg"></img>
      <p>
        Introducing the next leap forward in toaster technology! The Toast
        Master 5000 combines turbo charged industrial strength toast chambers
        with varied and sophisticated user settings. Toasting the perfect slice
        has never been easier!
      </p>
    </div>
  );
};

const reviewList = [
  '"I look forward to breakfast every morning now!"',
  "\"I've spent years looking for the perfect toaster, and now I've found it.\"",
  '"These toaster are the best! I don\'t go anywhere without one!"',
  '"Toaster Warehouse saved my marriage!"',
  '"These make the perfect Valentine\'s Day gift. Nothing is more romantic than toast!"',
  '"I\'ve been let down by alot of toasters so I was skeptical at first, but boy was I wrong!"',
  '"Toaster Warehouse are pioneers in the field of toasters."',
  '"When I need a toaster, I know who to call."',
  '"These toasters can\'t be beat!"',
  '"Only Toaster Warehouse brand toasters touch my counter top."',
  '"Busting out one of these beauties at a family gathering makes you the life of the party."',
  '"I thought I knew what was possible with toast. But my toast journey was just beginning!"',
  '"A couple slices and you\'ll understand why these guys are the best!"',
];

const currentRevs = [0, 1, 2];

function getRandom(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

function getUnique(arr) {
  if (arr.length === 0) {
    return getRandom(0, reviewList.length - 1);
  } else {
    let newIndex = getRandom(0, reviewList.length - 1);

    while (arr.includes(newIndex)) {
      newIndex = getRandom(0, reviewList.length - 1);
    }

    return newIndex;
  }
}

const Reviews = () => {
  const [review1, setReview1] = useState(reviewList[0]);
  const [review2, setReview2] = useState(reviewList[1]);
  const [review3, setReview3] = useState(reviewList[2]);

  useEffect(() => {
    setInterval(() => {
      const index = getUnique(currentRevs);
      currentRevs.splice(0, 1, index);
      setReview1(reviewList[index]);
    }, 5000);

    setInterval(() => {
      const index = getUnique(currentRevs);
      currentRevs.splice(1, 1, index);
      setReview2(reviewList[index]);
    }, 5000);

    setInterval(() => {
      const index = getUnique(currentRevs);
      currentRevs.splice(2, 1, index);
      setReview3(reviewList[index]);
    }, 5000);
  }, []);

  return (
    <div id="reviews">
      <h2 id="review-header" aria-label="review header">
        SATISFIED CUSTOMER REVIEWS
      </h2>
      <ul id="review-cont">
        <li id="review-column-1" className="review-column">
          {review1}
        </li>
        <li id="review-column-2" className="review-column">
          {review2}
        </li>
        <li id="review-column-3" className="review-column">
          {review3}
        </li>
      </ul>
    </div>
  );
};

const About = () => {
  return (
    <div id="about">
      <h2 id="about-header">ABOUT US</h2>
      <p id="about-text">
        Toaster Warehouse was first founded in 1892 in a small town called Loaf
        Shire. It's founder John Goodenbread had a saying,
        <strong id="motto">
          {" 'Reasonably quality products at reasonably affordable prices.' "}
        </strong>
        This motto continues to guide our company all these years later. From
        our warehouse to your counter, nothing brings the family together like
        toast!
      </p>
    </div>
  );
};
