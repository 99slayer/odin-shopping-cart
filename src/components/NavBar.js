import React from "react";
import '../styles/NavBar.css';
import { Link } from "react-router-dom";

export const NavBar = () => {
  return (
    <div id="nav-bar">
      <h1 id="heading">SHOPPING PROJECT</h1>
      <div id="nav-links">
        <Link to='/homepage'>HOME</Link >
        <Link to='/shoppage'>SHOP</Link>
      </div>
    </div>
  );
};