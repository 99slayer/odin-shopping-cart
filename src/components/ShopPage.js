import React from "react";
import { useOutletContext } from "react-router-dom";
import "../styles/ShopPage.css";

const order = [];

// Manages order array so cart items can be rendered in the order that they are added or removed.
const changeOrder = (list) => {
  for (const product in list) {
    if (order.includes(list[product].name) && list[product].count < 1) {
      const index = order.indexOf(list[product].name);
      order.splice(index, 1);
    } else if (order.includes(list[product].name)) {
      continue;
    } else if (list[product].count < 1) {
      continue;
    } else {
      order.push(list[product].name);
    }
  }
};

export const ShopPage = () => {
  const [productList, setProductList] = useOutletContext();

  const incrementProduct = (productNum) => {
    if (productList[productNum].count === 999) {
      return;
    }

    setProductList((prevList) => ({
      ...prevList,
      [productNum]: {
        ...prevList[productNum],
        count: parseInt(prevList[productNum].count + 1),
      },
    }));
  };

  const decrementProduct = (productNum) => {
    if (productList[productNum].count === 0) {
      return;
    }

    setProductList((prevList) => ({
      ...prevList,
      [productNum]: {
        ...prevList[productNum],
        count: parseInt(prevList[productNum].count - 1),
      },
    }));
  };

  const removeProduct = (productNum) => {
    setProductList((prevList) => ({
      ...prevList,
      [productNum]: { ...prevList[productNum], count: 0 },
    }));
  };

  // For dealing with user input.
  const handleProductInput = (productNum, e) => {
    if (e.target.value.length > 3 && e.target.key !== "Backspace") {
      return;
    }

    if (e.target.value[0] === "0") {
      const arr = e.target.value.split("");
      arr.splice(0, 1);
      e.target.value = arr.join("");
    }

    if (e.target.value === "") {
      e.target.value = "0";
    }

    setProductList((prevList) => ({
      ...prevList,
      [productNum]: {
        ...prevList[productNum],
        count: parseInt(e.target.value),
      },
    }));
  };

  const getTotal = (list) => {
    let total = 0;

    for (const product in list) {
      total += list[product].count * list[product].cost;
    }

    return total > 0 ? total.toFixed(2) : "";
  };

  return (
    <div id="shop-page">
      <ShoppingGrid
        increment={incrementProduct}
        decrement={decrementProduct}
        inputFunc={handleProductInput}
        list={productList}
      />
      <ShoppingCart
        increment={incrementProduct}
        decrement={decrementProduct}
        removeProduct={removeProduct}
        inputFunc={handleProductInput}
        totalFunc={getTotal}
        list={productList}
      />
    </div>
  );
};

const ShoppingGrid = (props) => {
  const { increment, decrement, inputFunc, list } = props;

  const renderCards = (obj) => {
    const elements = [];
    let i = 1;

    for (const product in obj) {
      elements.push(
        <ProductCard
          key={i}
          name={obj[product].name}
          cost={obj[product].cost}
          count={obj[product].count}
          image={obj[product].image}
          increment={increment}
          decrement={decrement}
          inputFunc={inputFunc}
          productNum={i}
        />
      );
      i += 1;
    }

    return elements;
  };

  return <div id="shopping-grid">{renderCards(list)}</div>;
};

const ProductCard = (props) => {
  const {
    name,
    cost,
    count,
    image,
    increment,
    decrement,
    inputFunc,
    productNum,
  } = props;

  const handleKeyDown = (e) => {
    const validInputs = [
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "0",
      "Backspace",
    ];

    if (!validInputs.includes(e.key)) {
      e.preventDefault();
    }
  };

  return (
    <div className="product-card">
      <img alt="" src={image} />
      <div className="product-card-infobar">
        <p className="product-card-name">{name}</p>
        <p className="product-card-cost">{"$" + cost.toFixed(2)}</p>
        <div className="product-card-count">
          <svg width={20} height={20}>
            <a onClick={() => decrement(productNum)}>
              <polygon className="decrement-btn" points="3,10 20,0 20,20" />
            </a>
          </svg>
          <input
            type="number"
            value={count}
            maxLength={3}
            onChange={(e) => {
              inputFunc(productNum, e);
            }}
            onKeyDown={handleKeyDown}
          ></input>
          <svg width={20} height={20}>
            <a onClick={() => increment(productNum)}>
              <polygon className="increment-btn" points="17,10 0,20 0,0" />
            </a>
          </svg>
        </div>
      </div>
    </div>
  );
};

const ShoppingCart = (props) => {
  const { increment, decrement, removeProduct, inputFunc, totalFunc, list } =
    props;

  const renderItems = (obj) => {
    changeOrder(list);
    const elements = [];
    let i = 1;

    while (elements.length !== order.length) {
      for (const product in obj) {
        if (obj[product].name !== order[i - 1]) {
          continue;
        } else {
          elements.push(
            <Item
              key={product}
              name={obj[product].name}
              cost={obj[product].cost}
              count={obj[product].count}
              increment={increment}
              decrement={decrement}
              removeProduct={removeProduct}
              inputFunc={inputFunc}
              productNum={product}
            />
          );

          i += 1;
        }
      }
    }

    return elements;
  };

  return (
    <div id="shopping-cart">
      <h2 id="cart-heading">CART</h2>
      <div id="item-cont">{renderItems(list)}</div>
      {/* can the getTotal function be moved to this component? */}
      <div
        id="cart-total"
        className={totalFunc(list) === "" ? "no-total" : ""}
      >{`TOTAL: $${totalFunc(list)}`}</div>
      <button id="checkout-btn">CHECKOUT</button>
    </div>
  );
};

const Item = (props) => {
  const {
    name,
    cost,
    count,
    increment,
    decrement,
    removeProduct,
    inputFunc,
    productNum,
  } = props;

  const handleKeyDown = (e) => {
    const validInputs = [
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "0",
      "Backspace",
    ];

    if (!validInputs.includes(e.key)) {
      e.preventDefault();
    }
  };

  return (
    <div className="item">
      <p className="item-name">{name}</p>
      <p className="item-cost">{"$" + (cost * count).toFixed(2)}</p>
      <div className="item-count-cont">
        <svg width={20} height={20}>
          <a onClick={() => decrement(productNum)}>
            <polygon className="decrement-btn" points="3,10 20,0 20,20" />
          </a>
        </svg>
        <input
          type="number"
          value={count}
          maxLength={3}
          onChange={(e) => {
            inputFunc(productNum, e);
          }}
          onKeyDown={handleKeyDown}
        ></input>
        <svg width={20} height={20}>
          <a onClick={() => increment(productNum)}>
            <polygon className="increment-btn" points="17,10 0,20 0,0" />
          </a>
        </svg>
      </div>
      <button
        type="button"
        className="item-delete-btn"
        onClick={() => removeProduct(productNum)}
      >
        X
      </button>
    </div>
  );
};
