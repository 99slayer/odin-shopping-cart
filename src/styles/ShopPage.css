input {
  /* should grow too */
  padding: 0px 4px 0px 4px;
  width: 40px;
  text-align: center;
}
#shop-page {
  height: 100%;
  display: flex;
  align-items: flex-start;
  gap: 10px;
  /* border: 2px solid black; */
}
#shopping-grid {
  /* setup grid overflow */
  flex: 3;
  height: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  justify-items: center;
  align-items: center;
  gap: 10px;
  /* border: 2px solid black; */
}
.product-card {
  height: 400px;
  width: 400px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid rgb(170, 170, 170);
  border-radius: 6px;
}
.product-card img {
  height: 100%;
  width: 100%;
  border-radius: 6px;
}
.product-card-infobar {
  display: flex;
  align-items: center;
  padding: 0px 10px 0px 10px;
  height: 50px;
  width: 380px;
  position: absolute;
  top: 337px;
  color: rgb(0, 1, 46);
  border: 1px solid rgb(196, 196, 196);
  border-radius: 6px;
  background-color: rgb(255, 255, 255);
}
.infobar-hover {
  /* display: flex;
  align-items: center; */
}
.product-card-name {
  font-size: 1.2em;
  font-weight: 600;
}
.product-card-cost {
  font-size: 1.2em;
  margin-left: auto;
}
.product-card-count {
  margin-left: 10px;
  display: flex;
  align-items: center;
  gap: 4px;
}
.increment-btn {
  fill: rgb(230, 230, 230);
  stroke: rgb(163, 163, 163);
  stroke-width: 1px;
}
.increment-btn:hover {
  cursor: pointer;
  fill: rgb(180, 180, 180);
}
.decrement-btn {
  fill: rgb(230, 230, 230);
  stroke: rgb(163, 163, 163);
  stroke-width: 1px;
}
.decrement-btn:hover {
  cursor: pointer;
  fill: rgb(180, 180, 180);
}
#shopping-cart {
  flex: 1;
  position: sticky;
  top: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  justify-content: stretch;
  border: 1px solid rgb(170, 170, 170);
  border-radius: 6px;
  background-color: white;
}
#cart-heading {
  display: flex;
  justify-content: center;
  align-items: center;
}
#item-cont {
  display: flex;
  flex-direction: column;
  gap: 10px;
}
#cart-total {
  font-size: 1.1em;
  margin: 20px 0px 10px 0px;
}
.no-total {
  display: none;
}
.item {
  padding: 4px;
  display: flex;
  border: 1px solid rgb(196, 196, 196);
  border-radius: 6px;
  justify-content: flex-start;
  align-items: center;
}
.item-info {
  flex: 1;
  display: flex;
  justify-content: space-between;
  gap: 20px;
}
.item-name {
}
.item-cost {
  margin-left: auto;
}
.item-count-cont {
  margin-left: 10px;
  display: flex;
  align-items: center;
  gap: 4px;
}
.item-delete-btn {
  margin-left: 4px;
  height: 26px;
  width: 26px;
  font-size: 1.2em;
  color: white;
  border: none;
  border-radius: 4px;
  background-color: rgb(255, 66, 66);
}
.item-delete-btn:hover {
  cursor: pointer;
  background-color: rgb(207, 66, 66);
}
