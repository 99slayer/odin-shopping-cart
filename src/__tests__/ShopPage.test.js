import React, { useState } from "react";
import { render, screen, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { MemoryRouter, Routes, Route, Outlet } from "react-router-dom";
import { ShopPage } from "../components/ShopPage";

const MockShopPage = () => {
  return (
    <MemoryRouter initialEntries={["/"]}>
      <Routes>
        <Route path="/" element={<MockApp />}>
          <Route index element={<ShopPage />} />
        </Route>
      </Routes>
    </MemoryRouter>
  );
};

const MockApp = () => {
  const [productList, setProductList] = useState({
    1: {
      name: "Toast Master 5000",
      cost: 125.99,
      count: 0,
      image: "toaster-1.jpg",
    },
    2: {
      name: "Toastie Classic",
      cost: 35.99,
      count: 0,
      image: "toaster-2.jpg",
    },
    3: {
      name: "Toastie Classic 2",
      cost: 40.0,
      count: 0,
      image: "toaster-3.jpg",
    },
    4: {
      name: "Toast Box",
      cost: 50.0,
      count: 0,
      image: "toaster-4.jpg",
    },
    5: {
      name: "Toast Box Supreme",
      cost: 70.0,
      count: 0,
      image: "toaster-5.jpg",
    },
    6: {
      name: "Toast Box Lite",
      cost: 35.99,
      count: 0,
      image: "toaster-6.jpg",
    },
    7: {
      name: "Industrial Toaster",
      cost: 200.0,
      count: 0,
      image: "toaster-7.jpg",
    },
    8: {
      name: "The Original",
      cost: 40.0,
      count: 0,
      image: "toaster-8.jpg",
    },
    9: {
      name: "Multi-Toast 5",
      cost: 85.0,
      count: 0,
      image: "toaster-9.jpg",
    },
  });

  return (
    <>
      <Outlet context={[productList, setProductList]} />
    </>
  );
};

describe("ShopPage component", () => {
  it("matches snapshot", () => {
    const shopPageComponent = renderer.create(<MockShopPage />).toJSON();
    expect(shopPageComponent).toMatchSnapshot();
  });

  describe("product card component", () => {
    it("product card increment button adds to product count", () => {
      render(<MockShopPage />);
      act(() =>
        userEvent.click(screen.getByTestId("product-card-increment-1"))
      );
      expect(screen.getByTestId("product-card-input-1").value).toBe("1");

      act(() =>
        userEvent.click(screen.getByTestId("product-card-increment-2"))
      );
      expect(screen.getByTestId("product-card-input-2").value).toBe("1");

      act(() =>
        userEvent.dblClick(screen.getByTestId("product-card-increment-3"))
      );
      expect(screen.getByTestId("product-card-input-3").value).toBe("2");
    });

    it("product card decrement button subtracts from count", () => {
      render(<MockShopPage />);
      act(() =>
        userEvent.dblClick(screen.getByTestId("product-card-increment-1"))
      );
      act(() =>
        userEvent.dblClick(screen.getByTestId("product-card-increment-1"))
      );
      act(() =>
        userEvent.click(screen.getByTestId("product-card-decrement-1"))
      );
      expect(screen.getByTestId("product-card-input-1").value).toBe("3");
    });
  });

  describe("cart item component", () => {
    beforeEach(() => {
      render(<MockShopPage />);
      act(() =>
        userEvent.click(screen.getByTestId("product-card-increment-1"))
      );
    });

    it("cart item to render after adding it via the product card", () => {
      expect(screen.getByTestId("cart-item-1")).toBeInTheDocument();
    });

    it("cart item to be removed after decrementing it below 1", () => {
      act(() =>
        userEvent.click(screen.getByTestId("product-card-decrement-1"))
      );
      expect(screen.queryByTestId("cart-item-1")).toBeNull();
    });

    it("cart item to be removed after deleting it via the item delete button", () => {
      act(() => userEvent.click(screen.getByTestId("item-delete-btn-1")));
      expect(screen.queryByTestId("cart-item-1")).toBeNull();
    });
  });
});
