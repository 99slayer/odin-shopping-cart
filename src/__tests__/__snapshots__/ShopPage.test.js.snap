// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopPage component matches snapshot 1`] = `
<div
  id="shop-page"
>
  <div
    id="shopping-grid"
  >
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-1.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toast Master 5000
        </p>
        <p
          className="product-card-cost"
        >
          $125.99
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-1"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-1"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-1"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-2.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toastie Classic
        </p>
        <p
          className="product-card-cost"
        >
          $35.99
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-2"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-2"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-2"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-3.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toastie Classic 2
        </p>
        <p
          className="product-card-cost"
        >
          $40.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-3"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-3"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-3"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-4.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toast Box
        </p>
        <p
          className="product-card-cost"
        >
          $50.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-4"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-4"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-4"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-5.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toast Box Supreme
        </p>
        <p
          className="product-card-cost"
        >
          $70.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-5"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-5"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-5"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-6.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Toast Box Lite
        </p>
        <p
          className="product-card-cost"
        >
          $35.99
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-6"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-6"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-6"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-7.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Industrial Toaster
        </p>
        <p
          className="product-card-cost"
        >
          $200.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-7"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-7"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-7"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-8.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          The Original
        </p>
        <p
          className="product-card-cost"
        >
          $40.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-8"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-8"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-8"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="product-card"
    >
      <img
        alt=""
        src="toaster-9.jpg"
      />
      <div
        className="product-card-infobar"
      >
        <p
          className="product-card-name"
        >
          Multi-Toast 5
        </p>
        <p
          className="product-card-cost"
        >
          $85.00
        </p>
        <div
          className="product-card-count"
        >
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-decrement-9"
              onClick={[Function]}
            >
              <polygon
                className="decrement-btn"
                points="3,10 20,0 20,20"
              />
            </a>
          </svg>
          <input
            data-testid="product-card-input-9"
            maxLength={3}
            onChange={[Function]}
            onKeyDown={[Function]}
            type="number"
            value={0}
          />
          <svg
            height={20}
            width={20}
          >
            <a
              data-testid="product-card-increment-9"
              onClick={[Function]}
            >
              <polygon
                className="increment-btn"
                points="17,10 0,20 0,0"
              />
            </a>
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    id="shopping-cart"
  >
    <h2
      id="cart-heading"
    >
      CART
    </h2>
    <div
      id="item-cont"
    />
    <div
      className="no-total"
      id="cart-total"
    >
      TOTAL: $
    </div>
    <button
      id="checkout-btn"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
